Error thrown when content files are in sub-folders.

### Describe the issue that you're seeing. Any Loom videos or screenshots usually help a lot!

This issue started happening after v1.5. Before it would just show directory under the file name in the CMS, now that Tina "supports" sub directories it actually throws an error when the collection is accessed in the CMS. 

I'm simply moving (or creating new) content fiels inside a sub folder. Then when I go to the CMS, it throws this error:
![image](https://github.com/tinacms/tinacms/assets/85646866/dca3ee4b-945c-4f97-890f-c25b3c20abbf)


And here's the log from the browser consol: 
```
index.es.js:3128 Error: Unable to fetch, errors: 
	Error querying file content/posts/05e33724787d3ffca5884867cef9de86d2808527.md from collection post. Please run "tinacms audit" or add the --verbose option  for more info
    at LocalClient.request (index.es.js:253:13)
    at async TinaAdminApi.fetchCollection (index.es.js:661:42)
    at async fetchCollection (index.es.js:3124:31)
fetchCollection @ index.es.js:3128
index.es.js:3170 Uncaught TypeError: Cannot read properties of undefined (reading 'name')
    at index.es.js:3170:79
    at invokePassiveEffectCreate (react-dom.development.js:23487:20)
    at HTMLUnknownElement.callCallback2 (react-dom.development.js:3945:14)
    at Object.invokeGuardedCallbackDev (react-dom.development.js:3994:16)
    at invokeGuardedCallback (react-dom.development.js:4056:31)
    at flushPassiveEffectsImpl (react-dom.development.js:23574:9)
    at unstable_runWithPriority (scheduler.development.js:468:12)
    at runWithPriority$1 (react-dom.development.js:11276:10)
    at flushPassiveEffects (react-dom.development.js:23447:14)
    at performSyncWorkOnRoot (react-dom.development.js:22269:3)
react-dom.development.js:20085 The above error occurred in the <GetCollection> component:

    at GetCollection (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:111449:3)
    at div
    at PageWrapper (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:111230:3)
    at GetCMS (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:111117:17)
    at CollectionListPage (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:111582:20)
    at div
    at div
    at div
    at Layout (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:110831:17)
    at DefaultWrapper (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:112820:3)
    at Routes (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:108055:5)
    at Router (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:108002:15)
    at HashRouter (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:108128:5)
    at CheckSchema (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:112693:3)
    at GetCMS (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:111117:17)
    at TinaAdmin (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:112713:3)
    at ErrorBoundary2 (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:110587:5)
    at AuthWallInner (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:109113:3)
    at div
    at ModalProvider (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:87898:24)
    at div
    at MutationSignalProvider (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:97453:33)
    at TinaUI (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:101549:17)
    at TinaCMSProvider (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:97352:3)
    at TinaProvider (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:101570:3)
    at BranchDataProvider (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:96241:3)
    at TinaCloudProvider (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:109177:28)
    at TinaCMSProvider2 (http://localhost:4001/admin/node_modules/.vite/deps/tinacms.js?v=38d0237e:110653:3)
    at TinaAdminWrapper
    at TinaEditProviderInner (http://localhost:4001/admin/node_modules/.vite/deps/tinacms_dist_edit-state.js?v=d12aab05:91:32)
    at EditProvider (http://localhost:4001/admin/node_modules/.vite/deps/chunk-NPNGPU2P.js?v=733f76ea:40:3)
    at TinaEditProvider (http://localhost:4001/admin/node_modules/.vite/deps/tinacms_dist_edit-state.js?v=d12aab05:19:3)
    at App

React will try to recreate this component tree from scratch using the error boundary you provided, ErrorBoundary2.
logCapturedError @ react-dom.development.js:20085
``` 
As you can see it's trying to fetch a file that does not exist (content/posts/05e33724787d3ffca5884867cef9de86d2808527.md). I noticed that the filename that it's trying to fetch is related to the sub-folder name. If I change the name, the hashed filename changes as well.

I've pinpointed the issue to this `Uncaught TypeError: Cannot read properties of undefined (reading 'name')`. It's trying to get the `name` of a `collection` which is undefinded. This is the source where it happens (Line: 58) https://github.com/tinacms/tinacms/blob/main/packages/tinacms/src/admin/components/GetCollection.tsx

From the looks of it it fails to `api.fetchCollection` for some reason so the `collection` is set to undefined. 

But I can't go further and would love it if you could take a look. This happens in my project and in a new, clean install of a next app and tina. I've done this several times and the issue is reproducable every time.

### Reproduction

https://github.com/arthuratmeetbreeze/tinacms-subfolder-issue-reproduction

### Steps to reproduce

1. Setup - Either create a new Next app then init Tina or use `npx create-tina-app@latest` then update to latest version of Tina (v1.5.5)
2. Create a sub-folder in `content/page`
3. Add a new page or duplicate one of the page and move into the sub-folder
4. Run `npm run dev`
5. Navigate to the Admin CMS --> Page Content

### System Info

```shell
System:
    OS: Windows 10 10.0.22621
    CPU: (16) x64 AMD Ryzen 7 6800H with Radeon Graphics
    Memory: 4.98 GB / 15.26 GB
  Binaries:
    Node: 18.14.2 - C:\Node\node.EXE
    npm: 9.5.0 - C:\Node\npm.CMD
  Browsers:
    Edge: Spartan (44.22621.1702.0), Chromium (113.0.1774.35)
    Internet Explorer: 11.0.22621.1
```


### Validations

- [X] Follow our [Code of Conduct](https://github.com/tinacms/tinacms/blob/main/CODE_OF_CONDUCT.md)
- [X] Read the [docs](https://tina.io/docs).
- [X] Check that there isn't [already an issue](https://github.com/tinacms/tinacms/issues) that reports the same bug.

--- a/.changeset/cuddly-steaks-share.md
+++ b/.changeset/cuddly-steaks-share.md
@@ -1,0 +1,5 @@
+---
+'@tinacms/graphql': patch
+---
+
+Fix windows bug in folder support

--- a/packages/@tinacms/graphql/src/database/datalayer.ts
+++ b/packages/@tinacms/graphql/src/database/datalayer.ts
@@ -14,7 +14,7 @@ import {
 } from './level'
 import { Collection } from '@tinacms/schema-tools'
 import path from 'path'
-import { normalizePath } from './utils'
+import { normalizePath } from './util'
 
 export enum OP {
   EQ = 'eq',

--- a/packages/@tinacms/graphql/src/database/datalayer.ts
+++ b/packages/@tinacms/graphql/src/database/datalayer.ts
@@ -14,6 +14,7 @@ import {
 } from './level'
 import { Collection } from '@tinacms/schema-tools'
 import path from 'path'
+import { normalizePath } from './utils'
 
 export enum OP {
   EQ = 'eq',
@@ -655,7 +656,7 @@ export class FolderTreeBuilder {
   }
 
   update(documentPath: string, collectionPath: string) {
-    let folderPath = path.dirname(documentPath)
+    let folderPath = path.dirname(normalizePath(documentPath))
     if (folderPath === '.') {
       folderPath = ''
     }
@@ -673,7 +674,7 @@ export class FolderTreeBuilder {
         if (!this._tree[current]) {
           this._tree[current] = new Set<string>()
         }
-        this._tree[current].add(path.join(current, part))
+        this._tree[current].add(normalizePath(path.join(current, part)))
         parent.push(part)
       })
     const current = parent.join('/')

